### DISCUSSION TITLE
How to serve a static page with a dynamic island?

### DISCUSSION THREAD

**Discussion by zoul** [vercel/next.js#59797] [Help]
### Summary

I have a page with a layout, inside the layout there is a session toolbar that contains the user avatar, if signed in, kind of like this:

```
<Page>
  <Layout>
    <SessionToolbar>
      <UserAvatarOrSignInLink/>
    </SessionToolbar>
  </Layout>
  <OtherContent/>
</Page>
```

Since there is no dynamic data on the rest of the page apart from the session toolbar, is it possible to make the page static and only load the session toolbar dynamically? Right now introducing the session toolbar (that uses `getServerSession` from NextAuth to read session info from the request) turns the whole page from ○ Static to λ Dynamic.

Thanks!

### Additional information

I’m still trying to wrap my head around React Server Components, Suspense and Streaming – which sounds like how this should or could work. But wrapping the session toolbar in a suspense and loading all data inside it didn’t seem to change anything, the route was still marked as Dynamic (whis means it’s not served from the CDN cache?).

**Comment by wilsonneto-swe** [comment]
I know that I`m not answering your question at all... I`m not fully updated with the new Next 14 features, but here we serve this kind of component as static and use a frontend local storage check to get the user session or show the link. Just one idea

Cheers!

**Reply by zoul** [reply]
Thank you! That’s what I ended up doing now – the page is static, the session toolbar is a client component and the session is read from cookies at the client side. But I was wondering if I could make it work with server components, reading the session at the server side while still serving the main page statically from the CDN. Because that scenario would then generalize well if I later needed a DB query or something that would be slower or impossible from the client.
